Replace $locationChangeSuccess in favor of optional customizable $roo…

elliottregan opened the pull request:
…tScope event.

kernowjoe commented:
I would suggest that the clean up doesn't need changing, and just adds to the cyclomatic complexity.

This would have the same effect
```javascript
!!rootScopeOnClose && rootScopeOnClose();
```
As this
```javascript
// remove event watcher
if (options.closeOn) {
    rootScopeOnClose && rootScopeOnClose();
}
```
There is no point checking if it should close it then checking if it has something to close then closing, may aswell skip the fist part.

The creation of the modal doesn't have the functionality that you described.
```javascript
if (options.closeOn) {
    rootScopeOnClose = $rootScope.$on(options.closeOn, cleanUpClose);
}
```
Maybe try change this to
```javascript
!options.hasOwnProperty('closeOn') && (options.closeOn = '$locationChangeSuccess');

if(options.closeOn !== false) {
    rootScopeOnClose = $rootScope.$on(options.onClose, cleanUpClose);
}
```
Then if nothing is specified it sticks to the standard way of working, unless you specifically tell it to change, and false turns it off completely.

This is all untested and just written in here, but *should* work...

elliottregan commented:
@kernowjoe Codeclimate wasn't a fan of those suggestions, but it works.

I changed the functionality in this commit to use `$locationChangeSuccess` as the default onClose event, which can be overridden by any value. If `false` is passed, nothing happens. I added that to the README.

elliottregan commented:
@dwmkerr What do you think the schedule would be to get this PR merged in? If there is anything I can do to help the process, just let me know.

dwmkerr commented:
Hi @elliottregan I'll look at it this week, really sorry about the delay am super crunched at work!!

elliottregan commented:
@dwmkerr Thanks!

lorenzodianni commented:
There are some news about this PR?

dwmkerr commented:
Sorry I'd let this one go stale, it looks like there's some conflicts in the PR now which'd have to be addressed first
